File: src/main/java/com/kakao/mobility/kmcoupon/dto/CouponResponse.java
package com.kakao.mobility.kmcoupon.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CouponResponse {
    private Long id;

    private BigDecimal useMinAmount;

    private BigDecimal discountAmount;

    private String usableFrom;

    private String usableUntil;

    private String status;
}

File: coupon-service-api/src/main/java/com/kakao/mobility/kmcoupon/dto/CouponResponse.kt
package com.kakao.mobility.kmcoupon.dto

import java.math.BigDecimal

data class CouponResponse(
    val id: Long,
    val useMinAmount: BigDecimal,
    val discountAmount: BigDecimal,
    val usableFrom: String,
    val usableUntil: String,
    val status: String
)